src/Aspire.Hosting/ApplicationModel/CustomResourceSnapshot.cs

```diff
@@ -159,6 +172,15 @@ internal void Deconstruct(out string name, out object? value, out bool isSensiti
 /// <param name="IsHighlighted">A flag indicating whether the command is highlighted in the UI.</param>
 public sealed record ResourceCommandSnapshot(string Type, ResourceCommandState State, string DisplayName, string? DisplayDescription, object? Parameter, string? ConfirmationMessage, string? IconName, IconVariant? IconVariant, bool IsHighlighted);
 
+/// <summary>
+/// A report produced by a health check about a resource.
+/// </summary>
+/// <param name="Name">The name of the health check that produced this report.</param>
+/// <param name="Status">The state of the resource, according to the report.</param>
+/// <param name="Description">An optional description of the report, for display.</param>
+/// <param name="Exception">An optional string containing exception details.</param>
+public sealed record HealthReportSnapshot(string Name, HealthStatus Status, string? Description, string? Exception);
```

Comment:
`Exception` -> `ExceptionDetails`? It is surprising to me to have a property called Exception that is a string and not an `System.Exception`.

Reply:
Agree. I've used `ExceptionText` elsewhere so will be consistent with that.

Reply:
This is partially done. Code in the app host like this snapshot is still `Exception`.